backend/de.metas.manufacturing/src/main/java/org/eevolution/model/validator/PP_Order.java

```diff
@@ -248,4 +256,21 @@ public void beforeDelete(@NonNull final I_PP_Order ppOrder)
 			deleteWorkflowAndBOM(ppOrderId);
 		}
 	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_CHANGE, ModelValidator.TYPE_BEFORE_NEW },
+			ifColumnsChanged = { I_PP_Order.COLUMNNAME_M_Product_ID, I_PP_Order.COLUMNNAME_PP_Product_BOM_ID })
+	public void validateBOMAndProduct(@NonNull final I_PP_Order ppOrder)
+	{
+		final ProductBOMId bomId = ProductBOMId.ofRepoId(ppOrder.getPP_Product_BOM_ID());
+
+		final ProductId productIdOfBOM = productBOMDAO.getBOMProductId(bomId);
+
+		if (ppOrder.getM_Product_ID() != productIdOfBOM.getRepoId())
+		{
+			throw new AdempiereException("PP_Order.M_Product_ID and PP_Order.PP_Product_BOM_ID.M_Product_ID don't match!")
```

Comment:
AFAIU the user can provoke this error from the UI, right? In that case we need an AD_Message.
:speaking_head: EN `The BOM product needs to match the manufacturing order's product` => DE `Das Produkt der Stückliste muss mit dem Produkt des Produktionsauftrags übereinstimmen`